basic/strdb.cc

```diff
@@ -184,9 +184,9 @@ void read_text(const char *file, int_func *func, StrDB &db, bool read_cached, bo
       }
       if(write_cached) logs("Writing to " << int_file);
 
-      char s[16384];
       char buf[16384]; int buf_i = 0; // Output buffer
-      while(in >> s) { // Read a string
+      for(std::string line; getline( in, line, ' ');) {
```

Comment:
1. To preserve 'breaking on whitespace functionality' the code can be modified as below.
`for(std::string line; getline(std::ws(in), line, ' ');)`

2. I think the reason why the previous code didn't work has got something to do with the internal memory allocation, buffer handling of the inputstream and copying to a char array of [16384] size. The recommended way to read is into a std::string rather than a char array. The stacktrace also seems to indicate that the internal memcpy operation didn't go fine.

I think an understanding beyond this may require me to dig deeper into the workings of the ifstream, limitations of reading via operator>> versus getline, and also into possibly libc library oeprations.